src/app/lib/profiles/profile.mainsail-migrator.ts

```diff
@@ -0,0 +1,71 @@
+import { IProfile, IProfileData, IProfileMainsailMigrator } from "./contracts.js";
+
+export class ProfileMainsailMigrator implements IProfileMainsailMigrator {
+	/**
+	 * Migrates the profile data from Mainsail to ArkVault if needed.
+	 *
+	 * @param {IProfileData} [data]
+	 * @return {Promise<IProfileData>}
+	 * @memberof Profile
+	 */
+	public async migrate(profile: IProfile, data: IProfileData): Promise<IProfileData> {
+		if (this.#requiresMigration(data)) {
+			data.wallets = await this.#migrateWallets(profile, data.wallets);
+		}
+
+		return data;
+	}
+
+	async #migrateWallets(profile: IProfile, wallets: IProfileData["wallets"]): Promise<IProfileData["wallets"]> {
+		const migratedWallets: IProfileData["wallets"] = {};
+
+		for (const [id, wallet] of Object.entries(wallets)) {
+			const migratedWallet = await this.#migrateWallet(profile, wallet);
+			migratedWallets[id] = migratedWallet;
+		}
+
+		return migratedWallets;
+	}
+
+	async #migrateWallet(
+		profile: IProfile,
+		wallet: IProfileData["wallets"][string],
+	): Promise<IProfileData["wallets"][string]> {
+		const migratedWallet: IProfileData["wallets"][string] = {
+			...wallet,
+			data: {
+				...wallet.data,
+				...(await this.#migrateWalletAddress(profile, wallet.data)),
+			},
+		};
+
+		return migratedWallet;
+	}
+
+	async #migrateWalletAddress(
+		profile: IProfile,
+		walletData: IProfileData["wallets"][string]["data"],
+	): Promise<IProfileData["wallets"][string]["data"]> {
+		const publicKey = walletData["PUBLIC_KEY"];
+		const wallet = await profile.walletFactory().fromPublicKey({ publicKey });
```

Comment:
this does not consider the scenario where the old profile has addresses with the same public key. In that case it should only import one time and not two. Scenario is:
- import an address on Mainnet, you get an `A...` address
- import an address on Devnet with the same mnemonic, you get a `D...` address

in the old wallet these are two separate instances, but now they share the same address and should be combined

<img width="621" height="282" alt="image" src="https://github.com/user-attachments/assets/d5197ba1-0232-4601-93f6-df190bfd529a" />